pubspec.yaml

```diff
@@ -5,7 +5,7 @@ homepage: 'https://supabase.io'
 repository: 'https://github.com/supabase/gotrue-dart'
 
 environment:
-  sdk: '>=2.12.0 <3.0.0'
+  sdk: '>=2.17.0 <3.0.0'
```

Comment:
Do we really need to increase this? The static method on `GenerateLinkType` can be created with `extensions` as well.

Reply:
This will enable other features we may want to use in the future that are only available in the latest dart versions. Also, this is a major version release, so the breaking change makes sense.

Reply:
Good call. I will bring it back to 2.12. I think other libraries use 2.17 as well, so I will look into them and bring them down to 2.12.

Reply:
I think the question is, how long do we want to support old Dart versions. I don't know if everyone is already using Flutter >=3.0.0, which doesn't ship Dart 1.17, right?

Reply:
Sorry @bdlukaa, some how I missed your message. The enhanced enum in 2.17.0 is enthusing and I personally use it all the time, but 2.17.0 was released only 5 months ago, and I feel like bumping the SDK version up to it, even though I first tried to do it in this PR, would cut off pretty good chunk of users. I think 2.14.0 is a good one to settle on, because it was released more than a year ago, and it at least has enhanced typedef feature.